### PR TITLE
Add internal solution for Korney Korneevich and XOR (easy version) (cf-1582F1)

### PR DESCRIPTION
**Pull request by ChuangSheep**
<del>WIP: an explanation will be added later</del> **Done**

_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making the changes.

### CODE REVIEW

**Comment by SansPapyrus683** [comment]
why dis draft

**Comment by bqi343** [comment]
@ChuangSheep Could you move this solution to `/solutions/gold`?

**Comment by ChuangSheep** [comment]
> @ChuangSheep Could you move this solution to `/solutions/gold/`?

sure, sorry didn't see the refactor

File: solutions/gold/cf-1582F1.mdx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+source: CF
+title: Korney Korneevich and XOR (easy version)
+author: Chuyang Wang
+---
+
+[Official Editorial](https://codeforces.com/blog/entry/96267)
+
+## Explanation
+
+To solve this problem, let's notice that the result of performing XOR
```
Comment (SansPapyrus683) [inline solutions/gold/cf-1582F1.mdx:12]:
```suggestion
To solve this problem, notice that the result of performing XOR
```

File: solutions/gold/cf-1582F1.mdx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+operations on a sequence of $a_i < 500$ is strictly less than $2^9$. That's
+because the ninth bit is the largest possible in the binary representation of
+$a_i$. 
+
+Let $dp[x][i]$ represent the feasibility of getting $x$ as a result of XOR of
```
Comment (SansPapyrus683) [inline solutions/gold/cf-1582F1.mdx:17]: _bit_ of an abrupt transition, don't you think? (sorry)

File: solutions/gold/cf-1582F1.mdx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+**Time Complexity:** $\mathcal{O}(n \cdot 2^{\lceil \log_2(\text{MAXA}) \rceil})$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+const int MAXA = 512;
```
Comment (SansPapyrus683) [inline solutions/gold/cf-1582F1.mdx:45]:
```suggestion
const int MAXA = 1 << 9
```
also i'd do `MAX_A`

File: solutions/gold/cf-1582F1.mdx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+int main() {
+	int n;
+	cin >> n;
+	vector<int> arr(n);
+	for (int &i : arr) {
+		cin >> i;
+	}
+
+	/*
+	 * dp[x][i] := whether it is possible to get x using XOR of an increasing
```
Comment (SansPapyrus683) [inline solutions/gold/cf-1582F1.mdx:56]: given that every other dp in code comments uses normal equal sign, i'd do away w/ the colon here (optional)

File: solutions/gold/cf-1582F1.mdx [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+	 */
+	vector<vector<bool>> dp(MAXA, vector<bool>(MAXA, false));
+	for (int i = 0; i < MAXA; i++) {
+		dp[0][i] = true;
+	}
+
+	for (int &a : arr) {
+		if (a == 0) {
+			continue;
+		}
```
Comment (SansPapyrus683) [inline solutions/gold/cf-1582F1.mdx:67]: haven't read the problem, but maybe explain this?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]
yeah

**Pull request merged**